client-spark/common/src/main/java/org/apache/spark/shuffle/RssSparkShuffleUtils.java

```diff
@@ -130,9 +130,13 @@ public static void applyDynamicClientConf(SparkConf sparkConf, Map<String, Strin
         sparkConfKey = RssSparkConfig.SPARK_RSS_CONFIG_PREFIX + sparkConfKey;
       }
       String confVal = kv.getValue();
-      if (!sparkConf.contains(sparkConfKey)
-          || RssSparkConfig.RSS_MANDATORY_CLUSTER_CONF.contains(sparkConfKey)) {
-        LOG.warn("Use conf dynamic conf {} = {}", sparkConfKey, confVal);
+      boolean isMandatory = RssSparkConfig.RSS_MANDATORY_CLUSTER_CONF.contains(sparkConfKey);
```

Comment:
Please add some test cases for this. 

BTW, it looks the RSS_MANDATORY_CLUSTER_CONF also should be allowed to configure by conf.

Reply:
This is existing, un-changed logic. This only improves logging distinguishing which conf is overridden and which is added due to dynamic conf. Before, both were logged with the same message on WARN level.

Reply:
Oh I misunderstand this. I think you just want to re-create client after fetching client conf. Right?

Reply:
Re-creating the client is one of the suggested solutions: #1554

I thought bootstrapping the config from the coordinator clients first thing would be better design. Re-creating the client looks a bit hacky.

Reply:
Got it.